scale_build/image/update.py

```diff
@@ -0,0 +1,158 @@
+
+    run_in_chroot('depmod', rootfs_logger, check=False)
+
+
+def custom_rootfs_setup(rootfs_logger):
+    # Any kind of custom mangling of the built rootfs image can exist here
+
+    # If we are upgrading a FreeBSD installation on USB, there won't be no opportunity to run truenas-initrd.py
+    # So we have to assume worse.
+    # If rootfs image is used in a Linux installation, initrd will be re-generated with proper configuration,
+    # so initrd we make now will only be used on the first boot after FreeBSD upgrade.
+    with open(os.path.join(CHROOT_BASEDIR, 'etc/default/zfs'), 'a') as f:
+        f.write('ZFS_INITRD_POST_MODPROBE_SLEEP=15')
+
+    run_in_chroot('update-initramfs -k all -u', logger=rootfs_logger)
+
+    # Generate native systemd unit files for SysV services that lack ones to prevent systemd-sysv-generator warnings
+    tmp_systemd = os.path.join(CHROOT_BASEDIR, 'tmp/systemd')
+    os.makedirs(tmp_systemd)
+    run_in_chroot(
+        '/usr/lib/systemd/system-generators/systemd-sysv-generator /tmp/systemd /tmp/systemd /tmp/systemd',
+        rootfs_logger
+    )
+    for unit_file in filter(lambda f: f.endswith('.service'), os.listdir(tmp_systemd)):
+        with open(os.path.join(tmp_systemd, unit_file), 'a') as f:
+            f.write(textwrap.dedent('''
+                [Install]
+                WantedBy=multi-user.target
+            '''))
+
+    for file_path in map(
+        lambda f: os.path.join(tmp_systemd, 'multi-user.target.wants', f),
+        filter(
+            lambda f: os.path.isfile(f) and not os.path.islink(f) and f != 'rrdcached.service',
+            os.listdir(os.path.join(tmp_systemd, 'multi-user.target.wants'))
+        )
+    ):
+        os.unlink(file_path)
+
+    run_in_chroot('rsync -av /tmp/systemd/ /usr/lib/systemd/system/')
+    shutil.rmtree(tmp_systemd, ignore_errors=True)
+
+
+def clean_rootfs(rootfs_logger):
+    to_remove = get_manifest()['base-prune']
+    run_in_chroot(
+        f'apt remove -y {" ".join(to_remove)}', rootfs_logger, f'Failed removing {", ".join(to_remove)!r} packages.'
+    )
+
+    # Remove any temp build depends
+    run_in_chroot('apt autoremove -y', rootfs_logger, 'Failed atp autoremove')
+
+    # We install the nvidia-kernel-dkms package which causes a modprobe file to be written
+    # (i.e /etc/modprobe.d/nvidia.conf). This file tries to modprobe all the associated
+    # nvidia drivers at boot whether or not your system has an nvidia card installed.
+    # For all truenas certified and truenas enterprise hardware, we do not include nvidia GPUS.
+    # So to prevent a bunch of systemd "Failed" messages to be barfed to the console during boot,
+    # we remove this file because the linux kernel dynamically loads the modules based on whether
+    # or not you have the actual hardware installed in the system.
+    if os.path.exists(os.path.join(CHROOT_BASEDIR, 'etc/modprobe.d/nvidia.conf')):
```

Comment:
If you use `with contextlib.suppress(FileNotFoundError)`, you won't have to repeat the same name twice